internal/staging/primitives/uint256/uint256_test.go

```diff
@@ -0,0 +1,4064 @@
+// Copyright (c) 2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package uint256
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// hexToBytes converts the passed hex string into bytes and will panic if there
+// is an error.  This is only provided for the hard-coded constants so errors in
+// the source code can be detected. It will only (and must only) be called with
+// hard-coded values.
+func hexToBytes(s string) []byte {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		panic("invalid hex in source file: " + s)
+	}
+	return b
+}
+
+// hexToUint256 converts the passed hex string into a Uint256 and will panic if
+// there is an error.  This is only provided for the hard-coded constants so
+// errors in the source code can be detected. It will only (and must only) be
+// called with hard-coded values.
+func hexToUint256(s string) *Uint256 {
+	if len(s)%2 != 0 {
+		s = "0" + s
+	}
+	b := hexToBytes(s)
+	if len(b) > 32 {
+		panic("hex in source file overflows mod 2^256: " + s)
+	}
+	return new(Uint256).SetByteSlice(b)
+}
+
+// randBigIntAndUint256 returns a big integer and a uint256 both created from
+// the same random value generated by the passed rng.
+func randBigIntAndUint256(t *testing.T, rng *rand.Rand) (*big.Int, *Uint256) {
+	var buf [32]byte
+	if _, err := rng.Read(buf[:]); err != nil {
+		t.Fatalf("failed to read random: %v", err)
+	}
+
+	// Create and return both a big integer and a uint256.
+	bigIntVal := new(big.Int).SetBytes(buf[:])
+	ui256Val := new(Uint256).SetBytes(&buf)
+	return bigIntVal, ui256Val
+}
+
+// TestUint256SetUint64 ensures that setting a scalar to various native integers
+// works as expected.
+func TestUint256SetUint64(t *testing.T) {
```

Comment:
It likely won't make any difference in practice since the test suite is already really fast, but do you think it makes sense to use `t.Parallel()` to allow the tests to run concurrently as a best practice unless there is a reason not to?

Reply:
I went ahead and modified all of the relevant commits to add it.  Like you said, it doesn't make much difference, but it does offer a speedup nonetheless:

```
Without: github.com/decred/dcrd/internal/staging/primitives/uint256      0.239s
With:    github.com/decred/dcrd/internal/staging/primitives/uint256      0.174s
```